Classes/Hooks/DataHandlerHook.php

```diff
@@ -54,10 +54,10 @@ public function processDatamap_afterDatabaseOperations($status, $table, $id, arr
                 ->execute()
                 ->fetch();
             if (!empty($record)) {
-                $timestamp = $record['publish_date'] ?? time();
+                $timestamp = (int) ($record['publish_date'] !== 0 ? $record['publish_date'] : time());
```

Comment:
Is `$record['publish_date']` always an integer? Otherwise, the `!== 0` condition will always evaluated being false.

Reply:
yes, the hook is running after typo3 did the database persistence so it decided to set the value to 0 if the field was empty.